src/github.com/getlantern/flashlight/statreporter/statreporter.go

```diff
@@ -147,11 +147,23 @@ func (r *reporter) run() {
 					categoryStats = make(stats)
 					dgAccum.categories[update.category] = categoryStats
 				}
-				switch update.action {
+				switch a := update.action.(type) {
```

Comment:
Yeah, it's pretty handy sometimes.  It's not as cool as the full on pattern matching that's available in some languages, but it's useful.